### ISSUE TITLE
Document Unique Id for Meetings originating from Exchange or Gmail

### ISSUE THREAD

**Issue by arturweiner** [OpenNTF/das-api-specs#10]
Hi Guys

I'm trying to integrate the domino calendar service in one of our web portals.

Usually we are getting back a list of events from the calendar service where the id property is something like "id":"D860077416285DBEC1258226003652FE-Lotus_Notes_Generated". With this I'm able to extract the document unique id and build a link to open the meeting in iNotes.

This works perfectly for meetings created in Notes. Unfortunately this is not valid for meetings which are comming from the outside world like Exchange or Gmail. For this kind of meetings the service is providing an id something like this: "id":"040000008200E00074C5B7101A82E00800000000004089BAF6D7D30100000000000000001000000004E6C04C43F3CA42AAFFAEB6928854FF".

After some research I've found that this is the id stored in the item $ICAL_UID of the meeting document. With this id i'm not able to open the meeting in iNotes anymore and an ugly "Entry not found in index" message page pops up.

I've also tried to open the event detail by GET/{folder}/{database}/api/calendar/events/{uid} but this is not providing an proper document unique id as well. Or do I miss something?

So can you tell me how I can get rid of my issue? Or is there an easy way to extend your service to fulfill my needs? Or maybe you guys could enhance your service with this useful property? An statement from your side would be appreciated! Thanks!

Regards

Artur

**Comment by ddelay** [comment]
Yes.  The JSON "id" property property maps to the UID property in iCalendar.  As you have discovered, it is different than the UNID in Notes.  When a meeting originates outside of Notes (e.g. Gmail), the iCalendar UID is completely up to the external product.  This makes sense when you consider that UID is the key for all scheduling workflow between heterogeneous products.  The originating product gets to decide the value of that key.

Unfortunately, the Domino calendar API doesn't directly expose the Notes UNID of an event.  So you aren't missing something.  We could consider this as a future enhancement, but it gets very tricky for repeating events.  In Notes, a repeating event sometimes corresponds to multiple documents -- each with it's own UNID.  In other words, UID-to-UNID _may be_ a one-to-many relationship.  

We have discussed this enhancement in the past, but the repeating event use case has always been the deal breaker.  Sorry.

**Comment by arturweiner** [comment]
Hi Dave

Thanks for your quick reply. Well this will make my boss unhappy... I have just one remark. You're saying that you have not implemented the Notes UNID as a json property because of repeating events... But on event documents originating from Notes it is exactly the Notes UNID in the id property just enriched with a post fix nobody is understanding for what this postfix is good for. Anyway, what I like to say isn't the argument of repeating events as the reason to not implement the Notes UNID a little bit contradictory? I mean the Notes UNID is already stored in the id property of Notes meetings so why not for the others I am asking my self. At the end of the day both are simple Notes documents isn't it? Well I do not have to understand everything on this whole wide world. Thanks!

Regards

Artur
